crate/client/src/kms_rest_client.rs

```diff
@@ -377,13 +378,26 @@ impl KmsRestClient {
     /// object. If the revocation reason is neither "key compromise" nor "CA
     /// compromise", the object is placed into the "deactivated" state, and the
     /// Deactivation Date is set to the current date and time.
-    pub async fn revoke(&self, request: Revoke) -> Result<RevokeResponse, RestClientError> {
+    pub async fn revoke(&self, request: Revoke) -> Result<RevokeResponse, ClientError> {
         self.post_ttlv::<Revoke, RevokeResponse>(&request).await
     }
 
+    /// This operation requests the server to send a message, which is a list of operations,
+    /// to the server.
+    ///The messages in the protocol consist of a message header, one or more batch items
+    /// (which contain OPTIONAL message payloads), and OPTIONAL message extensions.
+    /// The message headers contain fields whose presence is determined
+    /// by the protocol features used (e.g., asynchronous responses).
+    /// The field contents are also determined by whether the message is a request or a response.
+    /// The message payload is determined by the specific operation being requested
+    /// or to which is being replied.
+    pub async fn message(&self, request: Message) -> Result<MessageResponse, ClientError> {
+        self.post_ttlv::<Message, MessageResponse>(&request).await
+    }
```

Comment:
Avoid taking ownership here?
```suggestion
    pub async fn message(&self, request: &Message) -> Result<MessageResponse, ClientError> {
        self.post_ttlv::<Message, MessageResponse>(request).await
    }
```

Reply:
We probably could, bu the whole API is built around transfer of ownership. I would rather keep consistency here